### PR TITLE
Check If Quasar Is Active Using API

### PR DESCRIPTION
**Pull request by opticyclic**
To fix https://github.com/corda/corda/issues/5429

### CODE REVIEW

**Comment by roastario** [comment]
Can one of the admins verify this patch?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt [hunk 1]
```diff
@@ -708,7 +708,7 @@ class DriverDSLImpl(
             val effectiveP2PAddress = config.corda.messagingServerAddress ?: config.corda.p2pAddress
             return executorService.fork {
                 log.info("Starting in-process Node ${config.corda.myLegalName.organisation}")
-                if (!(ManagementFactory.getRuntimeMXBean().inputArguments.any { it.contains("quasar") })) {
+                if (!SuspendableHelper.isJavaAgentActive()) {
```
Comment (josecoll) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: The implementation of this method does not guarantee that the agent is active (as the boolean flag is set before agent argument checking which may lead to _IllegalStateException_)

Comment (josecoll) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: Could equally have used `JavaAgent.isActive()` which is clearer.

Comment (opticyclic) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: As per the bug report, it is using the same method as the StateMachine:
https://github.com/corda/corda/blob/release/os/4.3/node/src/main/kotlin/net/corda/node/services/statemachine/SingleThreadedStateMachineManager.kt#L309-L314

Comment (josecoll) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: Yes - both methods use the same underlying variable.
For clarity sake I prefer `JavaAgent.isActive()` to be used consistently.

Comment (opticyclic) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: So do you want me to change both instances to `JavaAgent.isActive()` in this pull request (even though the StateMachine isn't directly related to this issue)?

Comment (josecoll) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: Yes please.

**Review by josecoll — APPROVED** [review]
LGTM! 
Thanks for your contribution.

**Comment by josecoll** [comment]
@lockathan does this PR need to be forward merged ?

**Pull request merged**